packages/core/core-flows/src/product/steps/dismiss-product-variants-inventory.ts

```diff
@@ -0,0 +1,89 @@
+import { createStep, StepResponse } from "@medusajs/framework/workflows-sdk"
+import { ContainerRegistrationKeys, Modules } from "@medusajs/framework/utils"
+import { Link, Query } from "@medusajs/framework/modules-sdk"
+import { LinkDefinition } from "@medusajs/types"
+
+export const dismissProductVariantsInventoryStepId =
+  "dismiss-product-variants-inventory"
+
+export type DismissProductVariantsInventoryStepInput = {
+  variantIds: string[]
+}
+
+async function dismissVariantsInventory(
+  variantIds: string[],
+  query: Query,
+  link: Link
+): Promise<Record<string, string[]>> {
+  const dismissedVariantInventoryItems: Record<string, string[]> = {}
+
+  const { data: variantInventoryItems } = await query.graph({
+    entity: "product_variant_inventory_item",
+    fields: ["inventory_item_id", "variant_id"],
+    filters: {
+      variant_id: variantIds,
+    },
+  })
+
+  const dismissLinks: LinkDefinition[] = []
+  for (const variantId of variantIds) {
+    if (!variantId) {
+      continue
+    }
+
+    dismissedVariantInventoryItems[variantId] = variantInventoryItems
+      .filter((item) => item.variant_id === variantId)
+      .map((item: { inventory_item_id: string }) => {
```

Comment:
suggestion: `map` return a new array, which here we dont need as we just want to loop through. I would either suggest `forEach` or `for of` loop

Reply:
Sorry, it was a comment i thought i ve cancelled 🤣